pg_shard.c

```diff
@@ -813,6 +818,46 @@ ExtractFromExpressionWalker(Node *node, List **qualifierList)
 
 
 /*
+ * RemoveFromExpressionQualsMutator walks over a query and removes all quals
+ * from any FromExpr nodes it encounters. Returns a copy of the original query.
+ */
+static Node *
+RemoveFromExpressionQualsMutator(Node *originalNode, void *context)
```

Comment:
Correct. At the moment I'm removing _all_ local clauses and continue to send all clauses for remote execution.

#47 uses the word `VOLATILE` in its description, but the intent is closer to what you just described. Because the fix for what you describe needs to address the problem for `UPDATE` and `DELETE` statements as well (and possibly `INSERT`?) it will be better handled in a PR for #47 which will be focused entirely on that problem.